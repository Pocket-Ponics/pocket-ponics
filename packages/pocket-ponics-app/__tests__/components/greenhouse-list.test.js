import React from 'react'
import { shallow } from 'enzyme'
import Carousel from 'react-native-snap-carousel'

import GreenhouseList from '../../components/greenhouse/greenhouse-list'

describe('Greenhouse Display', () => {
	test('renders correctly', () => {
		const tree = shallow(
			<GreenhouseList
				greenhouses={[{},{},{}]}
				current={0}/>).html()
		expect(tree).toMatchSnapshot()
	})

	test('renders correctly', () => {
		const tree = shallow(
			<GreenhouseList
				greenhouses={[{},{},{}]}
				current={0}/>)


		expect(tree.find(Carousel).at(0).props().renderItem({ item: {
			tiers: []
		}})).toMatchSnapshot()

		expect(tree.find(Carousel).at(0).props().renderItem({ item: {
		}})).toMatchSnapshot()
	})
})