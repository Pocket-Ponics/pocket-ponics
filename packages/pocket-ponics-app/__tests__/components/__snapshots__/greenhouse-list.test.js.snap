// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greenhouse List renders correctly 1`] = `"<RCTScrollView decelerationRate=\\"fast\\" overScrollMode=\\"never\\" initialNumToRender=\\"6\\" maxToRenderPerBatch=\\"13\\" windowSize=\\"13\\" data=\\"[object Object],[object Object],[object Object]\\" firstItem=\\"0\\" itemWidth=\\"750\\" style=\\"opacity:0;width:750px;flex-direction:row\\" sliderWidth=\\"750\\" inactiveSlideScale=\\"0.95\\" inactiveSlideOpacity=\\"1\\" activeSlideAlignment=\\"start\\" activeAnimationType=\\"spring\\" activeAnimationOptions=\\"[object Object]\\" activeSlideOffset=\\"20\\" apparitionDelay=\\"0\\" autoplayDelay=\\"1000\\" autoplayInterval=\\"3000\\" callbackOffsetMargin=\\"5\\" containerCustomStyle=\\"[object Object]\\" contentContainerCustomStyle=\\"[object Object]\\" inactiveSlideShift=\\"0\\" layout=\\"default\\" lockScrollTimeoutDuration=\\"1000\\" loopClonesPerSide=\\"3\\" slideStyle=\\"[object Object]\\" swipeThreshold=\\"20\\" contentContainerStyle=\\"[object Object],[object Object]\\" scrollEventThrottle=\\"1\\" numColumns=\\"1\\" updateCellsBatchingPeriod=\\"50\\" viewabilityConfigCallbackPairs=\\"\\" stickyHeaderIndices=\\"\\"><View><View style=\\"0:[object Object]\\"></View><View style=\\"0:[object Object]\\"></View><View style=\\"0:[object Object]\\"></View></View></RCTScrollView>"`;

exports[`Greenhouse List renders correctly 2`] = `
<GreenhouseSwipeable
  greenhouse={
    Object {
      "tiers": Array [],
    }
  }
/>
`;

exports[`Greenhouse List renders correctly 3`] = `<NewGreenhouse />`;
